flintrock/ec2.py

```diff
@@ -310,12 +325,12 @@ def add_slaves(
                     ])
                 .create_tags(Tags=slave_tags))
 
-            existing_slaves = {i.public_ip_address for i in self.slave_instances}
+            existing_slaves = {i.public_ip_address or i.private_ip_address for i in self.slave_instances}
```

Comment:
It seems like if we have a new property to explicitly detect if we're running in a private subnet, `self.private_net`, we should use it here too (and just below) to switch between public and private addresses.

Reply:
Fixed.